### PR TITLE
docs: Update Documentation on Deny Policy Bug Fix

### LINKED ISSUE #15198: ClusterWideNetworkPolicy deny to CIDR not enforced
## Bug report

**General Information**

I am running a Kubernetes cluster on AWS and I want to restrict access to the AWS metadata service by default such that it is opt in. It seems that the `CiliumClusterwideNetworkPolicy` is not enforced when using `egressDeny` with `toCIDR`.

- Cilium version (run `cilium version`)
```
# cilium version
Client: 1.7.5 f524ca028 2020-06-12T14:10:36+02:00 go version go1.13.12 linux/amd64
Daemon: 1.7.5 f524ca028 2020-06-12T14:10:36+02:00 go version go1.13.12 linux/amd64
```
- Kernel version (run `uname -a`)
```
# uname -r
5.4.92-flatcar
```

Policy that is not getting enforced by cilium:
```
apiVersion: "cilium.io/v2"
kind: CiliumClusterwideNetworkPolicy
metadata:
  name: "block-aws-metadata-service"
spec:
  endpointSelector:
    matchExpressions:
      - key: k8s:io.kubernetes.pod.namespace
        operator: NotIn
        values:
          - kube-system
  egressDeny:
  - toCIDR:
    - 169.254.169.254/32
```

**How to reproduce the issue**

1. `kubectl apply` the "block-aws-metadata-service" policy to the cluster
2. exec into a pod that is not in kube-system and run `curl http://169.254.169.254`

I would expect that this request will time out, but it does succeed.

### PR DESCRIPTION
**Pull request by nathanjsweet**
docs: Update Documentation on Deny Policy Bug Fix

#15198 required Cilium to document the shortcoming
of CIDR deny policies. Since the bug has been fixed
this documentation is incorrect, it should be removed
with a remenant warning about the shortcoming in
prior versions.

Signed-off-by: Nate Sweet <nathanjsweet@pm.me>

### CODE REVIEW

**Comment by nathanjsweet** [comment]
blocked by #22966

File: Documentation/security/policy/language.rst [hunk 1]
```diff
@@ -1174,8 +1174,6 @@ for instructions.
 Deny Policies
 =============
 
-.. include:: ../../beta.rst
```
Comment (qmonnet) [inline Documentation/security/policy/language.rst]: Removing the beta status for a feature is not a negligible change that we should sneak in an otherwise unrelated commit. Can you please make a separate commit and give the motivations for the change?

**Review by qmonnet — CHANGES_REQUESTED** [review]
Commit log: s/remenant/remnant/

Do we need to keep the warning at all? The limitation should still be visible in earlier versions of the docs. I don't think we usually keep documenting former limitations after they've been addressed on newer versions?

**Comment by aditighag** [comment]
I'm moving this to draft as it's blocked on another PR. 

> blocked by #22966

File: Documentation/security/policy/language.rst [hunk 2]
```diff
@@ -1241,45 +1241,14 @@ Deny policies do not support: policy enforcement at L7, i.e., specifically
...
-      - toFQDNs:
-          - matchName: "www.google.com"
-
-.. _HostPolicies:
+Previous limitations and known issues
+-------------------------------------
+
+For Cilium versions prior to 1.14 deny-policies for peers outside the cluster
+sometimes did not work because of :gh-issue:`15198`.  Make sure that you are
+using version 1.13 or later if you are relying on deny policies to manage
```
Comment (qmonnet) [inline Documentation/security/policy/language.rst]: From the beginning of the paragraph, this should be `version 1.14 or later` instead of `1.13`, shouldn't it?

**Review by aditighag — COMMENTED** [review]
Same comment as Quentin.

**Review by joestringer — APPROVED** [review]
I'm fine with this. I agree that the version of the docs generated from this page will be for 1.14, so typically we don't document the behaviour of earlier versions in the newer docs.. but I also don't think it hurts to be a little more explicit. Sometimes users will read one version of the docs, then search for the corresponding section in a newer version and get confused about whether the issue still applies, because the newer docs don't have an explicit section to describe that the previous limitation no longer exists.

Presumably we'll want to strip out this statement at some point in future when we no longer support the older versions. That said, given that this text references `1.14`, it should be easy enough for us to find any such references in future in order to clean them up when we want to do so.

**Review by qmonnet — APPROVED** [review]
Thanks!

**Comment by joestringer** [comment]
@nathanjsweet does this need to be backported to older branches and maybe with the version edited, since https://github.com/cilium/cilium/pull/22966 was backported? Currently the v1.13 docs [present that known issue](https://docs.cilium.io/en/v1.13/security/policy/language/#limitations-and-known-issues) which is probably no longer a limitation?
